### PR TITLE
refactor table related components and add popover

### PR DESCRIPTION
**Pull request by mrahanjam**
# Refactored Table Components for ODF Tethering

## Description
This PR moves the table components to a higher level so that they can be shared among different tables in odf registration. It also adds the `action popover` to the `pending requests table`. I tried using already existing components but overriding `MUI styled` components with `styled` components turned out to not work properly. I have created [this](https://cdap.atlassian.net/browse/CDAP-18729) JIRA ticket to keep an eye on converting existing table components to `styled` components or upgrade MUI so that it's more compatible with `styled` components

## PR Type
- [ ] Bug Fix
- [X] Feature
- [ ] Build Fix
- [ ] Testing
- [X] General Improvement

## Links
Jira: [Jira issue #](fill-in.org)

## Screenshots
Screen Shot 2022-01-07 at 6 27 13 PM

### CODE REVIEW

**Review by seanfeld — APPROVED** [review]

File: app/cdap/components/Administration/TetheringTabContent/ActionPopover/index.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React from 'react';
+import styled, { css } from 'styled-components';
+import Popover from 'components/shared/Popover';
```
Comment (njbriggs) [inline app/cdap/components/Administration/TetheringTabContent/ActionPopover/index.tsx:19]: `components/shared/Popover` is built on `react-popper` and not Mui Popover. Could you use Mui `Popover` or `Dialog` for this component?

Comment (mrahanjam) [reply app/cdap/components/Administration/TetheringTabContent/ActionPopover/index.tsx:19]: @njbriggs I could, but the issue is adding the arrow to the top of the popover. Apparently only `Tooltip` and `Popper` support arrows inherently. `Tooltip` doesn't apply to our use case, and `Popper` with arrow is unnecessarily complicated compared to `react-popper` (same story with `Popover` in MUI). See [this](https://stackblitz.com/edit/poppers-with-arrows-x8a7ic?file=index.js) code sandbox for implementation. I can make it work if we've decided to stay away from react-popper but just wanted to make sure we're good with what MUI gives us.

Comment (mrahanjam) [reply app/cdap/components/Administration/TetheringTabContent/ActionPopover/index.tsx:19]: I put this together using MUI `Popover`. It's not 100% styled components but let me know what you think:

https://gist.github.com/mrahanjam/35c4e0819c14f72ea1d157dce750948d

Comment (njbriggs) [reply app/cdap/components/Administration/TetheringTabContent/ActionPopover/index.tsx:19]: We can stick with `react-popper` then.

**Review by njbriggs — APPROVED** [review]

**Pull request merged**